exchanges/alert/alert.go

```diff
@@ -18,66 +31,95 @@ type Notice struct {
 	wg sync.WaitGroup
 	// Segregated lock only for waiting routines, so as this does not interfere
 	// with the main calling lock, this acts as a rolling gate.
-	m sync.Mutex
+	mu sync.Mutex
+	// toActuatorRoutine is communication between the alert call and the
+	// actuator routine
+	toActuatorRoutine chan struct{}
+	// alerters are a pre allocated channel of communications pipes
+	alerters chan chan struct{}
 }
 
 // Alert establishes a state change on the required struct.
 func (n *Notice) Alert() {
 	// CompareAndSwap is used to swap from 1 -> 2 so we don't keep actuating
 	// the opposing compare and swap in method wait. This function can return
 	// freely when an alert operation is in process.
-	if !atomic.CompareAndSwapUint32(&n.sema, 1, 2) {
+	if !atomic.CompareAndSwapUint32(&n.sema, active, alerting) {
 		// Return if no waiting routines or currently alerting.
 		return
 	}
-	go n.actuate()
+
+	if n.toActuatorRoutine == nil {
```

Comment:
Should this be len instead of nil? Because the `actuate()` go routine will finish, but it won't be nil anymore.

Or should `actuate()` be `for { select {` ?

Reply:
Actuate() is persistent (never returns) so once the channel is set and the routine is scheduled it will continue operating for the life of the program.